### PR TITLE
Feature/uar 128 map and store existing managing officer data merged

### PR DESCRIPTION
**Pull request by AndreKramer7654**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-128

Adding MO mappings and re-work.

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/update/managing.officer.mapper.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import { isSameAddress } from "./mapper.utils";
+import { CompanyOfficer, FormerNameResource, DateOfBirth } from "@companieshouse/api-sdk-node/dist/services/company-officers/types";
+import { ManagingOfficerIndividual } from "../../model/managing.officer.model";
+import { ManagingOfficerCorporate } from "../../model/managing.officer.corporate.model";
+import { InputDate, yesNoResponse } from "../../model/data.types.model";
+
+export const mapToManagingOfficer = (officer: CompanyOfficer): ManagingOfficerIndividual => {
+  console.log(officer);
+  const raw = officer as any;
```
Comment (olaych) [inline src/utils/update/managing.officer.mapper.ts:9]: Do we still need raw?

File: src/controllers/update/beneficial.owner.type.controller.ts [hunk 2]
```diff
@@ -67,6 +75,22 @@ const getNextPage = (beneficialOwnerTypeChoices: BeneficialOwnerTypeChoice | Man
   }
 };
 
+export const retrieveManagingOfficers = async (req: Request, appData: ApplicationData) => {
+  const companyOfficers = await getCompanyOfficers(req, appData[EntityNumberKey] as string);
+  const session = req.session as Session;
+  if (companyOfficers !== undefined) {
```
Comment (bantao-ch) [inline src/controllers/update/beneficial.owner.type.controller.ts:81]: I would make this just 
            if (companyOfficers)

File: src/utils/update/managing.officer.mapper.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
+import { isSameAddress } from "./mapper.utils";
+import { CompanyOfficer, FormerNameResource, DateOfBirth } from "@companieshouse/api-sdk-node/dist/services/company-officers/types";
+import { ManagingOfficerIndividual } from "../../model/managing.officer.model";
+import { ManagingOfficerCorporate } from "../../model/managing.officer.corporate.model";
+import { InputDate, yesNoResponse } from "../../model/data.types.model";
+
+export const mapToManagingOfficer = (officer: CompanyOfficer): ManagingOfficerIndividual => {
+  console.log(officer);
```
Comment (bantao-ch) [inline src/utils/update/managing.officer.mapper.ts:8]: might want to remove the console.log

File: src/utils/update/managing.officer.mapper.ts (hunk 1 above)
Comment (AndreKramer7654) [reply src/utils/update/managing.officer.mapper.ts:9]: yes, for the id

File: src/utils/update/managing.officer.mapper.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        allFormerNames = holder;
+      } else {
+        allFormerNames += ", " + holder;
+      }
+    }
+    return allFormerNames;
+  }
+};
+
+export const mapAddress = (address: any) => {
```
Comment (stephanie-ch) [inline src/utils/update/managing.officer.mapper.ts:101]: This function looks the same as `mapAddress` in `mapper.utils`. Can it be removed?

File: src/controllers/update/beneficial.owner.type.controller.ts [hunk 5]
```diff
@@ -67,23 +68,34 @@ const getNextPage = (beneficialOwnerTypeChoices: BeneficialOwnerTypeChoice | Man
...
+        const individualBeneficialOwner = mapPscToBeneficialOwnerTypeIndividual(psc);
+        logger.info("Loaded individual Beneficial Owner " + individualBeneficialOwner.id + " is " + individualBenicitialOwner.first_name + ", " + individualBenifitialOwner.last_name);
+        const beneficialOwnerOther = mapPscToBeneficialOwnerOther(psc);
+        logger.info("Loaded Beneficial Owner Other " + beneficialOwnerOther.id + " is " + benicitialOwnerOther.name);
       } else if (psc.kind === "legal-person-with-significant-control") {
-        const beneficialOwnerGov: BeneficialOwnerGov = mapPscToBeneficialOwnerGov(psc);
-        setApplicationData(session, beneficialOwnerGov, BeneficialOwnerGovKey);
+        const beneficialOwnerGov = mapPscToBeneficialOwnerGov(psc);
+        logger.info("Loaded Beneficialicial Owner Gov " + beneficialOwnerGov.id + " is " + benicitialOwnerGov.name);
       }
```
Comment (cdrainCH) [inline src/controllers/update/beneficial.owner.type.controller.ts:98]: There are still spelling errors in the logger lines

File: src/utils/update/managing.officer.mapper.ts (hunk 4 above)
Comment (AndreKramer7654) [reply src/utils/update/managing.officer.mapper.ts:100]: yes - thought I did that

**Review by cdrainCH — APPROVED** [review]

**Review by olaych — APPROVED** [review]

**Pull request merged**